indico/modules/events/registration/__init__.py

```diff
@@ -76,8 +76,12 @@ def _inject_event_header(event, **kwargs):
                                           RegistrationForm.publish_registrations_enabled,
                                           _join=Registration.registration_form)
 
+    linked_participants = {reg.user: reg for reg in registrations if reg.user is not None}
+    non_linked_participants = set(reg for reg in registrations if reg.user is None)
+    participants = sorted(set(linked_participants.iterkeys()) | non_linked_participants, key=lambda p: p.id)
```

Comment:
`linked_participants.viewkeys()` already gives you a set-like object

`key=attrgetter('id')`